qiskit_experiments/framework/transpile_mixin.py

```diff
@@ -0,0 +1,98 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Transpile mixin class."""
+
+from __future__ import annotations
+from typing import Protocol
+
+from qiskit import QuantumCircuit, QuantumRegister, transpile
+from qiskit.providers import Backend
+
+
+class TranspileMixInProtocol(Protocol):
+    """A protocol to define a class that can be mixed with transpiler mixins."""
+
+    @property
+    def physical_qubits(self):
+        """Return the device qubits for the experiment."""
+
+    @property
+    def backend(self) -> Backend | None:
+        """Return the backend for the experiment"""
+
+    def circuits(self) -> list[QuantumCircuit]:
+        """Return a list of experiment circuits.
+
+        Returns:
+            A list of :class:`~qiskit.circuit.QuantumCircuit`.
+
+        .. note::
+            These circuits should be on qubits ``[0, .., N-1]`` for an
+            *N*-qubit experiment. The circuits mapped to physical qubits
+            are obtained via the internal :meth:`_transpiled_circuits` method.
+        """
+
+    def _transpiled_circuits(self) -> list[QuantumCircuit]:
+        ...
+
+
+class SimpleCircuitExtenderMixin:
```

Comment:
I wonder if we want to make this a feature of `BaseExperiment` rather than a mixin. I am not sure what the variable should be called but experiments could set something like `layout_and_translate_only = True` and if `BaseExperiment._transpiled_circuits` could just do the code here in that case.